src/System.Buffers/src/System/Buffers/DefaultArrayPool.cs

```diff
@@ -56,7 +62,17 @@ public override T[] Rent(int minimumLength)
 
             // Gettings here means we have too big of a request OR all the buckets from 
             // index through _buckets.Length are taken so we need to allocate a buffer on-demand.
-            return new T[Utilities.GetMaxSizeForBucket(index)];
+            int maxLength = Utilities.GetMaxSizeForBucket(_buckets.Length);
+            buffer = new T[Utilities.GetMaxSizeForBucket(index)];
+            if (log.IsEnabled())
+            {
+                if (buffer.Length > maxLength) 
+                    log.BufferOverMaximumAllocated(buffer.GetHashCode(), buffer.Length, maxLength);
+                else
+                    log.BufferAllocated(buffer.GetHashCode(), buffer.Length);
```

Comment:
Doesn't matter to me...I could remove the ```if (log.IsEnabled())``` check if you want. I put it in there to not violate the principle of least surprise (why do some logging statements show up all the time and some don't)